src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -898,33 +898,40 @@ InboundLedger::takeTxNode(
         return true;
     }
 
-    auto nodeIDit = nodeIDs.cbegin();
-    auto nodeDatait = data.begin();
-    TransactionStateSF filter(
-        mLedger->txMap().family().db(), app_.getLedgerMaster());
-
-    while (nodeIDit != nodeIDs.cend())
+    try
     {
-        if (nodeIDit->isRoot())
-        {
-            san += mLedger->txMap().addRootNode(
-                SHAMapHash{mLedger->info().txHash},
-                makeSlice(*nodeDatait),
-                snfWIRE,
-                &filter);
-            if (!san.isGood())
-                return false;
-        }
-        else
+        auto nodeIDit = nodeIDs.cbegin();
+        auto nodeDatait = data.begin();
```

Comment:
I realize we pass (and assume) same sized vectors but a better signature would be 
```
bool
InboundLedger::takeTxNode(
    std::vector<std::pair<SHAMapNodeID, Blob>> const& nodes,
    SHAMapAddNode& san)
```